### PR TITLE
Implements basic Memcache integration

### PR DESCRIPTION
**Pull request by danlamanna**
This implements the use of Memcache, currently only for the overview page.

I've still got to figure out how to get lib-memcache to work on Circle.

### CODE REVIEW

**Comment by jamiesnape** [comment]
Have you checked out this page?

https://cloud.google.com/appengine/docs/php/memcache/

Any use of Memcache is going to need to be compatible with GAE in future.

**Comment by danlamanna** [comment]
@jamiesnape Yes, assuming the `memcache->set` interface stays the same we should be fine. `connect` is the only "stubbed" function we're using, but we check for false return values so I think we're in the clear there as well.

File: include/memcache_functions.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
```
Comment (jamiesnape) [inline include/memcache_functions.php]: Add license header.

**Comment by jamiesnape** [comment]
The `composer.lock` is not correct. I will fix it.

**Comment by jamiesnape** [comment]
LGTM.

**Pull request merged**